File: app/src/main/java/ru/usedesk/sample/model/configuration/entity/Configuration.java
package ru.usedesk.sample.model.configuration.entity;

import androidx.annotation.NonNull;

public class Configuration {

    private final String urlChat;
    private final String urlOfflineForm;
    private final String urlToSendFile;
    private final String urlApi;
    private final String companyId;
    private final String accountId;
    private final String token;
    private final String clientEmail;
    private final String clientName;
    private final String clientPhoneNumber;
    private final String clientAdditionalId;
    private final String clientInitMessage;
    private final String customAgentName;
    private final Boolean foregroundService;
    private final Boolean withKnowledgeBase;

    public Configuration(
            @NonNull String urlChat,
            @NonNull String urlOfflineForm,
            @NonNull String urlToSendFile,
            @NonNull String urlApi,
            @NonNull String companyId,
            @NonNull String accountId,
            @NonNull String token,
            @NonNull String clientEmail,
            @NonNull String clientName,
            @NonNull String clientPhoneNumber,
            @NonNull String clientAdditionalId,
            @NonNull String clientInitMessage,
            @NonNull String customAgentName,
            @NonNull Boolean foregroundService,
            @NonNull Boolean withKnowledgeBase) {
        this.urlChat = urlChat;
        this.urlOfflineForm = urlOfflineForm;
        this.urlToSendFile = urlToSendFile;
        this.urlApi = urlApi;
        this.companyId = companyId;
        this.accountId = accountId;
        this.token = token;
        this.clientEmail = clientEmail;
        this.clientName = clientName;
        this.clientPhoneNumber = clientPhoneNumber;
        this.clientAdditionalId = clientAdditionalId;
        this.clientInitMessage = clientInitMessage;
        this.customAgentName = customAgentName;
        this.foregroundService = foregroundService;
        this.withKnowledgeBase = withKnowledgeBase;
    }

    @NonNull
    public String getUrlChat() {
        return urlChat;
    }

    @NonNull
    public String getUrlOfflineForm() {
        return urlOfflineForm;
    }

    @NonNull
    public String getUrlToSendFile() {
        return urlToSendFile;
    }

    @NonNull
    public String getUrlApi() {
        return urlApi;
    }

    @NonNull
    public String getCompanyId() {
        return companyId;
    }

    @NonNull
    public String getAccountId() {
        return accountId;
    }

    @NonNull
    public String getToken() {
        return token;
    }

    @NonNull
    public String getClientEmail() {
        return clientEmail;
    }

    @NonNull
    public String getClientName() {
        return clientName;
    }

    @NonNull
    public String getClientPhoneNumber() {
        return clientPhoneNumber;
    }

    @NonNull
    public String getClientAdditionalId() {
        return clientAdditionalId;
    }

    @NonNull
    public String getClientInitMessage() {
        return clientInitMessage;
    }

    @NonNull
    public String getCustomAgentName() {
        return customAgentName;
    }

    public boolean isForegroundService() {
        return foregroundService;
    }

    public boolean isWithKnowledgeBase() {
        return withKnowledgeBase;
    }
}

File: app/src/main/java/ru/usedesk/sample/model/configuration/entity/Configuration.kt

package ru.usedesk.sample.model.configuration.entity

import ru.usedesk.chat_sdk.entity.UsedeskChatConfiguration
import ru.usedesk.knowledgebase_sdk.entity.UsedeskKnowledgeBaseConfiguration

data class Configuration(
    val common: Common = Common(),
    val chat: Chat = Chat(),
    val kb: Kb = Kb()
) {
    data class Common(
        val materialComponents: Boolean = false,
        val urlApi: String = "https://secure.usedesk.ru",
        val apiToken: String = "",
        val clientEmail: String = "",
        val clientName: String = ""
    )

    data class Chat(
        val urlChat: String = "https://pubsubsec.usedesk.ru",
        val companyId: String = "",
        val channelId: String = "",
        val clientToken: String = "",
        val clientNote: String = "",
        val clientPhoneNumber: Long? = null,
        val clientAdditionalId: String? = null,
        val clientInitMessage: String = "",
        val clientAvatar: String? = null,
        val customAgentName: String = "",
        val messagesDateFormat: String = "",
        val messageTimeFormat: String = "",
        val messagesPageSize: Int = 20,
        val foregroundService: Boolean = false,
        val cacheFiles: Boolean = true,
        val adaptiveTimePadding: Boolean = true,
        val groupAgentMessages: Boolean = true,
        val additionalFields: Map<Long, String> = mapOf(),
        val additionalNestedFields: List<Map<Long, String>> = listOf()
    )

    data class Kb(
        val withKb: Boolean = true,
        val withKbSupportButton: Boolean = true,
        val noBackStack: Boolean = false,
        val kbId: String = "",
        val sectionId: Long? = null,
        val section: Boolean = false,
        val categoryId: Long? = null,
        val category: Boolean = false,
        val articleId: Long? = null,
        val article: Boolean = false
    )

    fun toChatConfiguration() = UsedeskChatConfiguration(
        urlChat = chat.urlChat,
        urlChatApi = common.urlApi,
        companyId = chat.companyId,
        channelId = chat.channelId,
        messagesPageSize = chat.messagesPageSize,
        clientToken = when (chat.clientToken) {
            "null" -> null
            else -> chat.clientToken
        },
        clientEmail = common.clientEmail,
        clientName = common.clientName,
        clientNote = chat.clientNote,
        clientPhoneNumber = chat.clientPhoneNumber,
        clientAdditionalId = chat.clientAdditionalId,
        clientInitMessage = chat.clientInitMessage,
        clientAvatar = chat.clientAvatar,
        cacheMessagesWithFile = chat.cacheFiles,
        additionalFields = chat.additionalFields,
        additionalNestedFields = chat.additionalNestedFields
    )

    fun toKbConfiguration() = UsedeskKnowledgeBaseConfiguration(
        urlApi = common.urlApi,
        accountId = kb.kbId,
        token = common.apiToken,
        clientEmail = common.clientEmail,
        clientName = common.clientName
    )
}